src/classes/storage.ts

```diff
@@ -114,16 +114,35 @@ export class Storage {
 
     if (!groundPlace || groundPlace.type !== placeType) {
       throw new Error(
-        `The ${placeType} with the Gpuid ${placeToRemoveGpuid} cannot be deleted because it cannot be found.`,
+        `The "${placeType}" with the Gpuid "${placeToRemoveGpuid}" cannot be deleted because it cannot be found.`,
       );
     }
 
     if (groundPlace.childs.length) {
       throw new Error(
-        `The ${placeType} with the Gpuid ${placeToRemoveGpuid} cannot be deleted because it is not empty.`,
+        `The "${placeType}" with the Gpuid "${placeToRemoveGpuid}" cannot be deleted because it has children.`,
       );
     }
 
+    // Delete the place from the GroundPlaces list.
     this.groundPlaces.splice(placeIndex, 1);
+
+    // If the place is a StopGroup, all references to this StopGroup are also removed from its possible StopCluster parent.
+    if (placeType === GroundPlaceType.GROUP) {
+      this.groundPlaces.reduce((reduceGroundPlaces: GroundPlace[], groundPlace: GroundPlace): GroundPlace[] => {
+        // Search through children from StopCluster that have a reference to the StopGroup deleted.
+        if (groundPlace.type === GroundPlaceType.CLUSTER && groundPlace.childs.includes(placeToRemoveGpuid)) {
+          const referenceIndex: number = groundPlace.childs.findIndex(
+            (stopGroupGpuid: StopGroupGpuid) => stopGroupGpuid === placeToRemoveGpuid,
+          );
+
+          groundPlace.childs.splice(referenceIndex, 1);
+        }
+
+        reduceGroundPlaces.push(groundPlace);
+
+        return reduceGroundPlaces;
```

Comment:
this seems weird to me, why not just use a `.map()` and `return groundPlace` at the end ?

Reply:
Indeed, no need to use a `reduce` here since the input table keeps the same shape as the output one, a `map` does the job very well, it's update!